contracts/deploy_advanced.sh

```
@@ -0,0 +1,358 @@
+    fi
+    
+    # Deploy contract
+    log_message "INFO" "Deploying $contract_name to $network..."
+    
+    local contract_id
+    contract_id=$(soroban contract deploy \
+        --wasm "$wasm_file" \
+        --source "$identity" \
+        --network "$network" 2>&1)
+    
+    if [ $? -eq 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

(SC2181)